<a href="https://colab.research.google.com/github/IBRAHIM354313/pemograman-lanjut/blob/main/CATATAN_Pengulangan_Musa_2C2230002.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**🖥️👩🏻‍💻PRAKTIKUM PEMROGRAMAN LANJUT🔥**
##**Pertemuan (2) PENGULANGAN (LOOP) - Senin, 21 Oktober 2024**

👤 Nama: Muhammad Musa Ibrahim / NRP: 2C2230002 / Kelas: III/A Sains Data

💬Deskripsi Pembahasan Materi Pengulangan:

1.1.  Pengulangan (LOOP) 🔂
- While Loop 🐋

- For Loop

1.2. Pengulangan Bersarang ( NESTED LOOP) 🔂🐦‍




#🤸🏻‍♀️ EKSPLORASI PRIBADI DI KELAS ✨

PENGENALAN LIST

In [1]:
my_list = [1, 2, 3]
my_list[0] = 10
print(my_list)

[10, 2, 3]


**Interpretasi Coding**: dari codingan diatas terdapat sebuah list bernama 'my_list' diinisialisasi dengan tiga elemen: [1, 2, 3] Kemudian di baris kedua 'my_list[0] = 10' mengubah elemen pertama (0) dari list 'my_list' menjadi 10. kemudian di baris ketiga 'print(my_list)' mencetak isi dari 'my_list' setelah perubahan dilakukan

**Interpretasi Output**: Setelah elemen pertama diubah dari 1 menjadi 10, list sekarang berisi:

elemen pertama menjadi 10 yang sebelumnya adalah 1
elemen kedua tetap 2 (tidak berubah)
elemen ketiga tetap 3 (tidak berubah)

PENGENALAN TUPLE

In [2]:
my_tuple = [1, 2, 3]
my_tuple[0] = 10  # Akan menghasilkan error karena tuple tidak bisa diubah
print(my_tuple)

[10, 2, 3]


Jenis Perulangan : For Loop

In [3]:
var_list = [1,2,3,4,5,6,7,8,9,10]
for i in var_list:
  print(i)


1
2
3
4
5
6
7
8
9
10


In [5]:
huruf = "FOOT BALL CLUB"
for i in huruf:
  print(i)

F
O
O
T
 
B
A
L
L
 
C
L
U
B


In [6]:
for i in range(15):
  print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14


In [7]:
for i in range (1000):
  print("Hello Baim Wong")

Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Baim Wong
Hello Ba

In [8]:
for i in range (2, 5, 1):
  print(i)

2
3
4


In [9]:
for i in range (7, 2, -1):
  print(i)

7
6
5
4
3


In [11]:
a = int( input( "Masukkan angka antara 1 hingga 10:  "))
b = int( input( "Masukkan angka antara 1 hingga 10:  "))

for i in range(a,b+1):
  print(i)

Masukkan angka antara 1 hingga 10:  3
Masukkan angka antara 1 hingga 10:  9
3
4
5
6
7
8
9


Perulangan While - Loop

In [13]:
counter = 2
while counter <= 7:
  print(counter)
  counter += 1

2
3
4
5
6
7


In [14]:
a = int( input())
b = int( input())

i = 10
while i<=20:
  print(i)
  i += 1

12
18
10
11
12
13
14
15
16
17
18
19
20


Perulangan: For Nested Loop

In [15]:
n = int(input())

for i in range(n):
  for j in range(n):
    print("*", end="")

10
****************************************************************************************************

Pembahasan Soal

In [16]:
#Kelipatan Terdekat

N = int(input("Masukkan N : "))

i = N + 1
while i % 5 != 0:
  i += 1

print("Output: ", i)

Masukkan N : 5
Output:  10


In [17]:
#Segitiga Bintang

N = int(input("Masukkan N : "))

for i in range (1, N+1):
  print("*" * i)

Masukkan N : 5
*
**
***
****
*****


In [18]:
# Urutan Kebawah

N = int(input("Masukkan N : "))

for i in range (N, 0, -1):
  for i in range (1, i+1):
    print(i, end="")
  print()

Masukkan N : 6
123456
12345
1234
123
12
1


In [22]:
#PIRAMIDA ANGKA

N = int(input("Masukkan nilai N: "))

# Cetak pola piramida angka
for i in range(1, N+1):
    print(' ' * (N-i) + ''.join(str(x) for x in range(1, 2*i)))

Masukkan nilai N: 6
     1
    123
   12345
  1234567
 123456789
1234567891011


In [21]:
# Kelipatan 2&3

N = int(input("Masukkan N : "))

#Loop dari 1 hingga N
for i in range (1, N+1):
  if i % 2 == 0 and i % 3 == 0:
    print("DuaTiga", end=" ")
  elif i % 2 == 0:
    print("Dua", end=" ")
  elif i % 3 == 0:
    print("Tiga", end=" ")
  else:
    print(i, end="")
print()

Masukkan N : 10
1Dua Tiga Dua 5DuaTiga 7Dua Tiga Dua 


* Interpretasi coding:

* Interpretasi Hasil:

   



#🧠 LATIHAN SOAL DI KELAS ⏱️

Problem (1) Membuat program menerima bilangan N dan menuliskan bilangan 10^x terkecil yang lebih dari N

In [ ]:
N = int(input("Masukkan N : "))
pangkat = 0

while 10 ** pangkat <= N:
  pangkat += 1

print("Output: ", 10 ** pangkat)

Masukkan N : 1234
Output:  10000


 Interpretasi coding:
1. Input program diatas untuk memasukkan nilai N
2. variabel "pangkat" diinisialisasi dengan 0
3. perulangan menggunakan while dikarenakan kondisi yang tidak pasti. artinya
   program tersebut tidak tahu sebelumnya berapa kali perulangan harus dilakukan
4. Perulangan akan terus berjalan selama hasil dari 10 pangkat masih lebih
  kecil dari N. ketika sudah melewati atau sama dengan N, perulangan akan berhenti.

  
 Interpretasi Hasil:
* Masukkan N : 5
* output : 10
* Masukkan N : 1234
* output : 10000

Problem (2) Membuat program menerima bilangan N dan membuat pola segitiga angka dengan dasar N.

In [ ]:
N = int(input("Masukkan N: "))

for i in range(1, N + 1): #Loop luar baris
    for j in range(1, i + 1): #Loop dalam baris
        print(j, end=" ")
    print()

- Interpretasi coding:
1. Pada program ini, kita menggunakan nested loop karena pada output yang menghasilkan segitiga angka memiliki jumlah angka yang bertambah 1.
2. Nested loop dapat mengatur jumlah pengulangan pada setiap baris
3. Loop luar baris berguna untuk mengulang baris 1 hingga baris N
4. Loop dalam baris mengulang dari kolom 1 hingga kolom i
4. print (j, end="") berguna untuk mencetak angka j tanpa pindah baris
5. print() berguna untuk pindah ke baris baru setelah setiap baris selesai dicetak
- Interpretasi Hasil:
* 1
* 1 2
* 1 2 3
* 1 2 3 4
* 1 2 3 4 5







Program (3) Kalkulator Sederhana

In [24]:
# Kelipatan 10 terkecil
def kelipatan_10_terkecil(N):
    # Mencari kelipatan 10 terkecil yang lebih besar dari N
    kelipatan = 10
    while kelipatan <= N:
        kelipatan += 10
    return kelipatan

# Menerima input dari user
N = int(input("Masukan N: "))

# tampilkan hasil
hasil = kelipatan_10_terkecil(N)
print(hasil)


Masukan N: 5
10


**Interpretasi coding**: Di baris pertama, terdapat sebuah fungsi 'kelipatan10_terkecil' di definisikan dengan satu parameter, yaitu 'N'.

'kelipatan = 10' variabel 'kelipatan' diinisialisasi dengan nilai 10, yang merupakan kelipatan 10 pertama.

kemudian terdapat loop: while kelipatan <= N: kelipatan += 10. italicized text loop 'while' ini akan terus berjalan selama nilai 'kelipatan' *kurang dari atau sama dengan *'N' di dalam loop, nilai 'kelipatan' ditambah 10 pada setiap iterasi.

'return kelipatan' setelah loop selesai fungsi ini mengembalikan nilai 'kelipatan', yaitu kelipatan 10 terkecil yang lebih besar dari 'N'.

hasil dari fungsi 'kelipatan_10_terkecil' *disimpan dalam variabel *'hasil', kemudian dicetak ke layar.

**Interpretasi Hasil**: Fungsi 'kelipatan_10_terkecil(5)' kemudian dipanggil.

Proses perhitungan di dalam fungsi adalah sebagai berikut:

Inisialisasi kelipatan dengan nilai 10. Memeriksa apakah kelipatan (10) kurang dari atau sama dengan N (5). Karena kondisi ini tidak terpenuhi, loop tidak dijalankan. Setelah loop selesai (yang tidak dijalankan dalam kasus ini), fungsi mengembalikan nilai kelipatan, yaitu 10. Sehingga, output akhir menunjukkan bahwa kelipatan 10 terkecil yang lebih besar dari 5 adalah 10, dan itulah yang dicetak sebagai hasil.

Program (4) Bilangan Ganjil

In [25]:
awal = int(input("Masukkan angka awal: "))
akhir = int(input("Masukkan angka akhir: "))
x = []
y = []

for i in range (awal, akhir+1):
  if i % 2 == 0:
    x.append(i)
  else:
    y.append(i)

print("Bilangan Ganjil: ", y)
print("Bilangan Genap: ", x)

Masukkan angka awal: 5
Masukkan angka akhir: 6
Bilangan Ganjil:  [5]
Bilangan Genap:  [6]


**Interpretasi coding:**

fungsi 'segitiga_angka' di definisikan dengan satu parameter, yaitu 'n'.

pada baris ketiga terdapat loop 'for i in range(1, n + 1)' loop ini mengulang dari 1 hingga n, di mana setiap iterasi merepresentasikan satu baris dari segitiga angka

kemudian terdapat loop for j in range(1, i+1): print(j, end='') loop ini mengulang dari 1 hingga 'i', di mana 'i' adalah nomor baris saat ini. pada setiap irerasi, angka 'j' di cetak tanpa berpindah ke baris baru (karena penggunaan 'end=''), sehingga semua angka pada baris yang sama akan di cetak bersebelahan.

'segitiga_angka' fungsi tsb dipanggil dengan argumen 'n' yang telah dimasukkan oleh pengguna

 **Interpretasi Hasil:**

Fungsi segitiga_angka(5) kemudian dipanggil dan menghasilkan pola segitiga angka sebagai berikut: Baris pertama (i=1): Mencetak angka 1. Baris kedua (i=2): Mencetak angka 1 dan 2. Baris ketiga (i=3): Mencetak angka 1, 2, dan 3. Baris keempat (i=4): Mencetak angka 1, 2, 3, dan 4. Baris kelima (i=5): Mencetak angka 1, 2, 3, 4, dan 5.

#🏆 POST TEST 🎯

Problem (1) membuat program jika "x" kelipatan dari a maka keluarkan "Siap". Jika "x" kelipatan dari b maka keluarkan "bang". jika "x" kelipatan c maka keluarkan "jago". jika "x" bukan merupakan kelipatan seluruhnya "else", maka tampilkan angka aja.

In [ ]:
N = int(input("Masukkan N : "))
a = int(input("Masukkan A : "))
b = int(input("Masukkan B : "))
c = int(input("Masukkan C : "))

for x in range (1, N+1):
  if x % a == 0 and x % b == 0 and x % c == 0:
    print("SiapBangJago", end=" ")
  elif x % a == 0 and x % b == 0:
    print("SiapBang", end=" ")
  elif x % a == 0 and x % c == 0:
    print("SiapJago", end=" ")
  elif x % b == 0 and x % c == 0:
    print("BangJago", end=" ")
  elif x % a == 0:
    print("Siap", end=" ")
  elif x % b == 0:
    print("Bang", end=" ")
  elif x % c == 0:
    print("Jago", end=" ")
  else:
    print(x, end=" ")
print()

- Interpretasi coding:
1. Input program N, a, b, c
2. Menggunakan pengulangan for loop untuk mengulang angka dari 1 hingga N
3. pada saat menggunakan if elif else, kita menjabarkan dari tiap masing masing angka, apabila kelipatan x habis dibagi a, b, c maka akan keluar output "SiapBangJago"
4. Apabila kelipatan x habis dibagi a dan b maka akan keluar output "SiapBang"
5. Apabila kelipatan x habis dibagi b dan c maka akan keluar output "BangJago"
6. Apabila kelipatan x habis dibagi a maka akan keluar output "Siap"
7. Apabila kelipatan x habis dibagi b maka akan keluar output "Bang"
8. Apabila kelipatan x habis dibagi c maka akan keluar output "Jago"
9. Apabila kelipatan x tidak dapat dibagi a,b,c maka akan keluar output bilangan itu sendiri "N"

- Interpretasi hasil:

Masukkan nilai N = 12 Masukkan Nilai a = 2 Masukkan Nilai b = 3 Masukkan Nilai c = 4 Output : 1 Siap Bang SiapJago 5 SiapBang 7 SiapJago Bang Siap 11 SiapBangJago



#🏆 POST TEST 🎯#

Problem (1)

In [26]:
def siap_bang_jago(N, A, B, C):
    # Iterasi dari 1 sampai N
    for x in range(1, N + 1):
        output = ""

        # Jika x kelipatan dari A, tambahkan 'Siap' ke output
        if x % A == 0:
            output += "Siap"

        # Jika x kelipatan dari B, tambahkan 'Bang' ke output
        if x % B == 0:
            output += "Bang"

        # Jika x kelipatan dari C, tambahkan 'Jago' ke output
        if x % C == 0:
            output += "Jago"

        # Jika output kosong, cetak angkanya saja
        if output == "":
            print(x)
        else:
            print(output)

# Input nilai N, A, B, C dari user
N = int(input("Masukkan nilai N: "))
A = int(input("Masukkan nilai A: "))
B = int(input("Masukkan nilai B: "))
C = int(input("Masukkan nilai C: "))

# Panggil fungsi
siap_bang_jago(N, A, B, C)

Masukkan nilai N: 12
Masukkan nilai A: 6
Masukkan nilai B: 3
Masukkan nilai C: 1
Jago
Jago
BangJago
Jago
Jago
SiapBangJago
Jago
Jago
BangJago
Jago
Jago
SiapBangJago


In [34]:
def ok1_bang_jago(N, A, B, C):
    # Iterasi dari 1 sampai N
    for x in range(1, N + 1):
        output = ""

        # Jika x kelipatan dari A, tambahkan 'Siap' ke output
        if x % A == 0:
            output += "ok"

        # Jika x kelipatan dari B, tambahkan 'Bang' ke output
        if x % B == 0:
            output += "Bang"

        # Jika x kelipatan dari C, tambahkan 'Jago' ke output
        if x % C == 0:
            output += "Jago"

        # Jika output kosong, cetak angkanya saja
        if output == "":
            print(x)
        else:
            print(output)

# Input nilai N, A, B, C dari user
N = int(input("Masukkan nilai N: "))
A = int(input("Masukkan nilai A: "))
B = int(input("Masukkan nilai B: "))
C = int(input("Masukkan nilai C: "))

# Panggil fungsi
ok_bang_jago(N, A, B, C)


Masukkan nilai N: 10
Masukkan nilai A: 5
Masukkan nilai B: 2
Masukkan nilai C: 1
Jago
BangJago
Jago
BangJago
okJago
BangJago
Jago
BangJago
Jago
okBangJago


**Interpretasi coding**:

Fungsi siap_bang_jago(N, A, B, C):

1.**Fungsi ini akan melakukan iterasi dari angka 1 hingga N**

Setiap angka diperiksa apakah merupakan kelipatan dari A, B, atau C.

Jika angka tersebut adalah kelipatan dari A, maka akan ditambahkan kata "Siap" ke variabel output.

Jika angka tersebut adalah kelipatan dari B, maka akan ditambahkan kata "Bang".

Jika angka tersebut adalah kelipatan dari C, maka akan ditambahkan kata "Jago".

Jika suatu angka bukan kelipatan dari A, B, atau C, maka angka itu sendiri yang akan ditampilkan.

Setelah pengecekan selesai untuk setiap angka, hasilnya dicetak: baik berupa angka itu sendiri atau gabungan kata "Siap", "Bang", "Jago" tergantung kondisi kelipatannya.

2.**Iterasi Angka (Looping dari 1 sampai N)**:

Program akan melakukan pengecekan pada setiap angka, mulai dari 1 hingga N. Untuk setiap angka, program akan melihat apakah angka tersebut bisa dibagi habis (kelipatan) oleh A, B, atau C

3.**Pengecekan Kelipatan**:
Penggunaan operator modulus (%) digunakan untuk memeriksa apakah suatu angka merupakan kelipatan dari A, B, atau C.

Contoh: Jika x % A == 0, artinya x adalah kelipatan dari A

4.**Penggabungan Output**:
Jika suatu angka adalah kelipatan dari lebih dari satu bilangan (misalnya kelipatan A dan B), maka kedua kata yang sesuai akan digabungkan dalam output, contohnya: "SiapBang"

5.**Input**:
Program meminta pengguna untuk memasukkan nilai N (batas iterasi) serta angka A, B, dan C

**Interpretasi hasil:**

**Test Case 1:**

 Angka 1: Bukan kelipatan dari A, B, atau C, jadi program mencetak angka 1.

Angka 2: Kelipatan dari A (2), maka program mencetak "Siap".

Angka 3: Kelipatan dari B (3), maka program mencetak "Bang".

Angka 4: Kelipatan dari A dan C (4), maka program mencetak "SiapJago".

Angka 5: Bukan kelipatan dari A, B, atau C, jadi program mencetak angka 5.

Angka 6: Kelipatan dari A (2) dan B (3), maka program mencetak "SiapBang".

Angka 7: Bukan kelipatan dari A, B, atau C, jadi program mencetak angka 7.

Angka 8: Kelipatan dari A (2) dan C (4), maka program mencetak "SiapJago".

Angka 9: Kelipatan dari B (3), maka program mencetak "Bang".

Angka 10: Kelipatan dari A (2), maka program mencetak "Siap".

Angka 11: Bukan kelipatan dari A, B, atau C, jadi program mencetak angka 11.

Angka 12: Kelipatan dari A (2), B (3), dan C (4), maka program mencetak "SiapBangJago"

**Test Case 2:**

 setiap angka dari 1 hingga 5 adalah kelipatan dari A, B, dan C (semua angka adalah kelipatan dari 1), maka setiap angka dari 1 hingga 5 akan mencetak "SiapBangJago".

📌Catatan:
  - Notebook python ini dibuat untuk memudahkan temen-temen belajar ngoding dan pengumpulan tugas.
  - Silakan download notebook ini lalu upload di Google Collab milik kalian agar format `pertemuan`, `topik pertemuan`,
`tanggal`, `deskripsi pembahasan materi`, `baris coding` dan `interpretasi` bisa disesuaikan dengan kebutuhan teman-teman.

  - Interpretasi coding dan hasil dimaksudkan agar teman-teman semakin memahami apa yang dipelajari.

  - Deskripsi pembahasan materi dimaksudkan agar teman-teman semakin memahami apa fokus yang dipelajari.

  - Tulisan di setiap sebelum baris code disebut `markdown`. Ada beberapa jenis simbol yang memudahkan kepenulisan `markdown` kita, yaitu seperti : untuk mengatur ukuran tulisan biasanya tulisan pada header menggunakan ini`#`, `##`, `###`, untuk membuat pointer menggunakan ini `-`, untuk membuat tulisan ter-highlight menggunakan ini *` `* , untuk short-cut membuat tulisan jadi **bold** gunakan ini `** **`, dan untuk *tulisan miring* gunakan ini `* *` .

  - Semangat belajar! 🌟✨

©️ LABORATORIUM SAINS DATA 2024